docs/features.rst

```diff
@@ -19,70 +18,111 @@ Storing features
 Finally, they specify how the feature matrix is stored with ``storage_type`` (currently ``numpy`` or ``lilcom``),
 and where to find it with the ``storage_path``. In the future there might be more storage types.
 
+.. code-block:: yaml
+
+    - channel_id: 0
+      duration: 16.04
+      num_features: 23
+      num_frames: 1604
+      recording_id: recording-1
+      start: 0.0
+      storage_path: test/fixtures/libri/storage/dc2e0952-f2f8-423c-9b8c-f5481652ee1d.llc
+      storage_type: lilcom
+      type: fbank
+
+
+Creating custom feature extractor
+*********************************
+
+There are two components needed to implement a custom feature extractor: a configuration and the extractor itself.
+We expect the configuration class to be a dataclass, so that it can be automatically mapped to dict and serialized.
+The feature extractor should inherit from :ref:`lhotse.features.FeatureExtractor`,
+and implement a small number of methods/properties.
+The base class takes care of initialization (you need to pass a config object), serialization to YAML, etc.
+A minimal, complete example of adding a new feature extractor:
+
+.. code-block::
+
+    @dataclass
+    class ExampleFeatureExtractorConfig:
+        frame_shift: Seconds = 0.01
+
+    class ExampleFeatureExtractor(FeatureExtractor):
+        name = 'example-feature-extractor'
+        config_type = ExampleFeatureExtractorConfig
+
+        def extract(self, samples: np.ndarray, sampling_rate: int) -> np.ndarray:
+            f, t, Zxx = stft(samples, sampling_rate, noverlap=round(self.frame_shift * sampling_rate))
+            return np.abs(Zxx)
+
+        @property
+        def frame_shift(self) -> Seconds:
+            return self.config.frame_shift
+
+The overridden members include:
+
+- ``name`` for easy debuggability/automatic re-creation of an extractor;
+- ``config_type`` which specifies the complementary configuration class type;
+- ``extract()`` where the actual computation takes place;
+- ``frame_shift`` property, which is key to know the relationship between the duration and the number of frames.
+
+Additionally, there are two extra methods than when overridden, allow to perform dynamic feature-space mixing (see Cuts):
+
+.. code-block::
+
+    @staticmethod
+    def mix(features_a: np.ndarray, features_b: np.ndarray, gain_b: float) -> np.ndarray:
+        raise ValueError(f'The feature extractor\'s "mix" operation is undefined.')
+
+    @staticmethod
+    def compute_energy(features: np.ndarray) -> float:
+        raise ValueError(f'The feature extractor\'s "compute_energy" is undefined.')
+
+They are:
+
+- ``mix()`` which specifies how to mix two feature matrices to obtain a new feature matrix representing the sum of signals;
+- ``compute_energy()`` which specifies how to obtain a total energy of the feature matrix, which is needed to mix two signals with a specified SNR.
```

Comment:
Can you be more specific about 'gain' and the return value of compute_energy, e.g. must they be in dB (and can you give an example of how dB is computed, because I think there are energy and magnitude versions).